packages/api/src/services/lsp-state.ts

```diff
@@ -0,0 +1,127 @@
+import * as uma from "@uma/sdk";
+import { BatchRead, toString, toNumber, parseBytes, nowS } from "../libs/utils";
+import { AppState } from "..";
+import { lsps } from "../tables";
+
+type Instance = uma.clients.lsp.Instance;
+type Config = undefined;
+type Dependencies = Pick<
+  AppState,
+  "lsps" | "registeredLsps" | "provider" | "collateralAddresses" | "shortAddresses" | "longAddresses" | "multicall"
+>;
+
+export default (config: Config, appState: Dependencies) => {
+  const { lsps, registeredLsps, provider, collateralAddresses, shortAddresses, longAddresses, multicall } = appState;
+
+  // default props we want to query on contract
+  const staticProps: [string, (x: any) => any][] = [
+    ["collateralPerPair", toString],
+    ["priceIdentifier", parseBytes],
+    ["collateralToken", toString],
+    ["longToken", toString],
+    ["shortToken", toString],
+    ["finder", toString],
+    ["financialProductLibrary", toString],
+    ["customAncillaryData", toString],
+    ["prepaidProposerReward", toString],
+  ];
+
+  const dynamicProps: [string, (x: any) => any][] = [
+    ["expiryPrice", toString],
+    ["expiryPercentLong", toString],
+    ["expirationTimestamp", toNumber],
+    ["contractState", toNumber],
+  ];
+
+  async function batchRead(calls: [string, (x: any) => any][], instance: Instance, address: string) {
+    const result = await BatchRead(multicall)(calls, instance);
+    return {
+      address,
+      updated: nowS(),
+      ...result,
+    };
+  }
+  async function updateLsp(address: string, startBlock?: number | "latest", endBlock?: number) {
```

Comment:
this contains complex logic to optimize how often we call infura based on things like, have we seen this contract before, was it expired or active, or transitioning from expired to active.